src/threadNames.h

```diff
@@ -0,0 +1,33 @@
+#ifndef _THREADNAME_H
+#define _THREADNAME_H
+
+#include <jvmti.h>
+#include <pthread.h>
+#include <map>
+#include <string>
+#include "vmStructs.h"
+#include "vmEntry.h"
+
+
+typedef std::map<int, std::string> NamesMap;
+
+class ThreadNames {
+  private:
+    pthread_mutex_t _lock;
+    NamesMap _storage;
+    jclass _threadClass;
+    jfieldID _eetop;
+  public:
+    static ThreadNames _instance;
```

Comment:
I'm afraid keeping the same map throughout the lifetime of the application may have some undesired side effects. Java thread name will be tied to the given PID forever, even if PID reused weeks later for some native thread. Also this may lead to excessive memory usage if threads keep borning and dying.

Reply:
I agree. I will create new map and register callbacks every time when profiler is started and unregister callbacks when it stops.